apps/web/src/app/advertisers/campaigns/page.tsx

```diff
@@ -1,28 +1,163 @@
 'use client';
 import { buttonVariants } from '@/components/ui/button';
 import { formatDate } from '@/util/dates';
-import { Campaign } from '@repo/types';
+import { Campaign, Network } from '@repo/types';
+import { set } from 'date-fns';
 import Link from 'next/link';
 import { useEffect, useState } from 'react';
 export default function CampaignsPage(): JSX.Element {
 	const [campaigns, setCampaigns] = useState<Campaign[]>([]);
+	const [cpms, setCpms] = useState<number[]>([]);
+	const [cpcs, setCPCs] = useState<number[]>([]);
+	const [ctrs, setCtrs] = useState<number[]>([]);
+	const [spends, setSpends] = useState<number[]>([]);
+	const [impressions, setImpressions] = useState<number[]>([]);
+	const [clicks, setClicks] = useState<number[]>([]);
+	const [networkNames, setNetworkNames] = useState<string[]>([]);
```

Comment:
So one of the main things here is that the user doesn't need to interact with this page dynamically. The only place for the user to click is the `Link` component. This means that this entire thing can be a server component. Which also means no need to use `useState` or `useEffect` either. All the data fetching logic can go directly in the body of the functional component.